src/main/java/SnippetManager.java

```diff
@@ -0,0 +1,120 @@
+        if (snippoFolder.mkdir()) {
+            System.out.println("Folder created: " + snippoFolder.getName());
+        }
+
+        for (final File fileEntry : Objects.requireNonNull(snippoFolder.listFiles())) {
+            if (!fileEntry.isDirectory()) {
+                Integer snippetId = Integer.parseInt(fileEntry.getName());
+                Snippet loadedSnippet = JsonIO.getInstance().loadSnippetFromJson(generatePathToSnippetJson(snippetId));
+                snippets.put(snippetId, loadedSnippet);
+            }
+        }
+    }
+
+    private String generatePathToSnippetJson(Integer snippetId) {
+        return pathToSnippoDir + "/" + snippetId;
+    }
+
+    String listSnippets(Map<Integer, Snippet> snippetsToList) {
+        StringBuilder response = new StringBuilder("All snippetsToList: \n");
+
+        for (Integer snippetId: snippetsToList.keySet()) {
+            response.append("Id: "+snippetId+"\n").append(snippetsToList.get(snippetId).toString()).append("\n");
+        }
+
+        return response.toString();
+    }
+
+    String listAll() {
+        return listSnippets(snippets);
+    }
+
+    Integer create(String title, String content, String language, String[] tags) {
+        Integer newSnippetId = getNextId();
+        snippets.put(
+                newSnippetId,
+                new Snippet(generatePathToSnippetJson(newSnippetId), title, content, language, tags)
+        );
+
+        return newSnippetId;
+    }
+
+    String read(Integer id) {
+        if (isValidId(id)) {
+            return snippets.get(id).toString();
+        } else {
+            onSnippetNotFound();
+            return "";
+        }
+    }
+
+    void delete(Integer id) {
+        if (isValidId(id)) {
+            if (new File(generatePathToSnippetJson(id)).delete()) {
+                snippets.remove(id);
+            } else {
+                System.out.println("Cannot delete the snippet and its file. Id: " + id);
+            }
+        } else {
+            onSnippetNotFound();
+        }
+    }
+
+    void edit(Integer id) {
+        if (isValidId(id)) {
+            Snippet snippetToEdit = snippets.get(id);
+            snippetToEdit.edit();
+        } else {
+            onSnippetNotFound();
+        }
+    }
+
+    Map<Integer, Snippet> filter(String wordToContain, String language, String[] tags) {
+        return snippets.entrySet()
+                .stream()
+                .filter(
+                        entry -> entry.getValue().getTitle().contains(wordToContain) ||
+                                entry.getValue().getContent().contains(wordToContain)
+                )
+                .filter(entry -> language.equals("") ||
+                        entry.getValue().getLanguage().equals(language))
+                .filter(entry -> (tags.length == 1 && tags[0].equals("")) ||
+                        Arrays.asList(entry.getValue().getTags()).containsAll(Arrays.asList(tags)))
```

Comment:
This is a lot to unpack, function on function on function. Maybe any way to do this differently?